octoprint_slicer/static/js/slicer.js

```diff
@@ -499,6 +506,42 @@ $(function() {
             self.renderer.render( self.scene, self.camera );
         };
 
+	self.slicerProperties = ko.observable();
+	self.updateSlicerProperties = function(newSlicersArray) {
+	    if (_.isArray(newSlicersArray) &&
+		newSlicersArray.length > 0) {
+		OctoPrint
+		    .get(OctoPrint.getBlueprintUrl("slicer") +
+			 "slicer/properties")
+		    .then(function (result) {
+			self.slicerProperties(result);
+		    });
+	    } else {
+		self.slicerProperties({});
+	    }
+	}
+	self.slicingViewModel.slicers.subscribe(self.updateSlicerProperties);
+
+	self.sameDevice = ko.computed(function() {
```

Comment:
Obviously you know a lot more about inner working of OctoPrint than I do @eyal0 !

I might have misunderstood what `same_device` in OctoPrint is, but I estimate for most OctoPrint users (like me) `same_device` is always `true`. 

If so, I'd propose a 4th option: drop `sameDevice` from this PR. For most users, it'll still function perfectly. For the few users whose `same_device` is some times `false`, they won't be able to slice a model when they are printing on a different Raspberry Pi, but it's probably not end of the world for them.

When your PR #1748 is accepted, we can just add a tiny bit of code to make even that use case go away.

The reason why I propose this option is to see if there is a way to use only 20% of the code to make 80% of the use cases work. But please let me know if I'm wrong on the "20%" part. For example, if you will often run into this problem (not being able to slice when you should), my proposal won't fly.

And yes there is a way to specify the minimum version of OctoPrint this plugin requires.